File: BetterTridents/src/main/java/project/kompass/btk/BTK.java
package project.kompass.btk;

import project.kompass.btk.listener.*;
import org.bukkit.plugin.PluginManager;
import org.bukkit.plugin.java.JavaPlugin;

public class BTK extends JavaPlugin {

    @Override
    public void onEnable() {
        PluginManager pm = getServer().getPluginManager();

        // Register all separated listener classes
        pm.registerEvents(new TridentAttributeListener(), this);
        pm.registerEvents(new TridentRiptideListener(), this);
        pm.registerEvents(new TridentChannelingListener(), this);
        pm.registerEvents(new TridentDamageListener(), this);
        pm.registerEvents(new TridentLootingListener(), this);
        pm.registerEvents(new TridentAnvilListener(), this);
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
    }
}

File: BetterTridents/src/main/kotlin/project/kompass/btk/BTK.kt
package project.kompass.btk

import project.kompass.btk.listener.*
import org.bukkit.plugin.java.JavaPlugin

class BTK : JavaPlugin() {

    override fun onEnable() {
        saveDefaultConfig()

        val pm = server.pluginManager

        pm.registerEvents(TridentAttributeListener(), this)
        pm.registerEvents(TridentRiptideListener(), this)
        pm.registerEvents(TridentChannelingListener(this), this)
        pm.registerEvents(TridentDamageListener(), this)
        pm.registerEvents(RangedLootingListener(), this)
        pm.registerEvents(TridentAnvilListener(), this)
        pm.registerEvents(SpearListener(), this)
        pm.registerEvents(ArmorDurabilityListener(this), this)
        pm.registerEvents(PotionSoupStackListener(), this)
        pm.registerEvents(BundleCapacityListener(this), this)
        pm.registerEvents(AlwaysEatListener(this), this)
        pm.registerEvents(EnchantablePetArmorListener(), this)
        pm.registerEvents(WindChargeListener(this), this) // Registered Wind Charge Listener

        // Initiate Copper Armor hazard loops & particle task
        val copperArmorListener = CopperArmorListener()
        pm.registerEvents(copperArmorListener, this)
        copperArmorListener.startArmorCheckTask(this)
        copperArmorListener.startParticleTask(this)

        // Initiate Magical Campfire task
        val campfireListener = MagicalCampfireListener(this)
        pm.registerEvents(campfireListener, this)
        campfireListener.startCampfireTask()
    }

    override fun onDisable() {
    }
}